legend-depot-server/src/main/java/org/finos/legend/depot/server/LegendDepotServer.java

```diff
@@ -42,19 +44,24 @@ public static void main(String... args) throws Exception
         new LegendDepotServer().run(args);
     }
 
-
     @Override
     protected GuiceBundle<DepotServerConfiguration> buildGuiceBundle()
     {
-
         return GuiceBundle.defaultBuilder(DepotServerConfiguration.class)
                 .modules(new InfoPageModule())
                 .modules(new DepotServerModule())
                 .modules(new DepotServerResourcesModule())
+                .modules(new PureModelContextModule())
                 .modules(new StoreMongoModule())
                 .modules(new ReadOnlyServicesModule())
                 .modules(new TracingModule())
                 .build();
     }
 
+    @Override
+    public void run(DepotServerConfiguration configuration, Environment environment)
+    {
+        super.run(configuration, environment);
+        environment.jersey().register(LegendDepotServerJacksonProvider.class);
```

Comment:
I'm not sure I follow, this `JacksonJsonProvider` does provide that differentiating logic. If it sees a PMCD, it will use a dedicated object mapper with all subtypes registered, else, it will use the default object mapper used everywhere else. Now as for the default object mapper, the only thing we configure is the date format for serializing Date fields.

Actually, what I'm doing here by extending `JacksonJaxbJsonProvider` is my attempt to follow closest to what we had before. If you think we should break from that then sure, I have removed the dependency on `JacksonJaxbJsonProvider`, else everything stays the same